src/bsp/include/bsp/ui.h

```diff
@@ -0,0 +1,29 @@
+#ifndef __UI_H_
+#define __UI_H_
+
+/**
+ * @brief Manages the user interface
+ * The user interface can consist of buttons, LED, serial print ports, etc.
+ * It is not a graphical interface.
+ * The board will communicate it's state via the UI and the user can send commands via the UI.
+ */
```

Comment:
> Naming convention: If it is an interface, it should be named it accordingly, iUI or IUI (seems awkward in this case since is short, could be renamed UserInterface for clarity). For the filename, I would also use CamelCase to match what we use for class names.

For the filename, if we go camel case, all the files will go camel case. I have no opinion or preference on that for once, just ask Charles and Sam IRL what they think about it and I will change it if you conclude that camelcase is preferable for the filenames. Just small point though, if we do change, do we change the generated C files too? Or accept that the C files are in snake case and C++ in camelcase?

Reply:
True. Either PascalCase or CamelCase, I don't really care, I just dislike snake_case for file names. Again, we need coding standard definition :laughing: